$NetBSD: patch-ag,v 1.43 2026/05/04 23:58:58 taca Exp $

* Add common support for NetBSD.
* Only define HAS_DB if it hasn't been defined.
* Add support for FreeBSD 11 and 12.

--- src/util/sys_defs.h.orig	2026-05-01 18:55:47.000000000 +0000
+++ src/util/sys_defs.h
@@ -40,6 +40,7 @@
     || defined(NETBSD4) || defined(NETBSD5) || defined(NETBSD6) \
     || defined(NETBSD7) | defined(NETBSD8) || defined(NETBSD9) \
     || defined(NETBSD10) || defined(NETBSD11) || defined(NETBSD12) \
+    || defined(NETBSD) \
     || defined(EKKOBSD1) || defined(DRAGONFLY)
 #define SUPPORTED
 #include <sys/param.h>
@@ -52,7 +53,9 @@
 #define DEF_MAILBOX_LOCK "flock, dotlock"
 #define HAS_SUN_LEN
 #define HAS_FSYNC
+#ifndef HAS_DB
 #define HAS_DB
+#endif
 #define HAS_SA_LEN
 #define NATIVE_DB_TYPE	"hash"
 #if (defined(__NetBSD_Version__) && __NetBSD_Version__ >= 104250000)
@@ -1263,6 +1266,8 @@ extern int h_errno;
 #define DEF_SMTP_CACHE_DEMAND	0
 #endif
 
+#define RECEIVED_ENVELOPE_FROM
+
  /*
   * We're not going to try to guess like configure does.
   */
